### PR TITLE
Bump to 23.3.0.

### PR DESCRIPTION
**Pull request by jezdez**
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [ ] Bumped the build number (if the version is unchanged)
* [x] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by jezdez** [comment]
@conda-forge-admin, please rerender

File: recipe/meta.yaml [hunk 1]
```diff
@@ -1,29 +1,33 @@
 {% set name = "conda-libmamba-solver" %}
-{% set version = "23.1.0" %}
+{% set version = "23.3.0" %}
 
 package:
   name: {{ name|lower }}
   version: {{ version }}
 
 source:
   url: https://github.com/conda/conda-libmamba-solver/archive/{{ version }}.tar.gz  # temporary
```
Comment (jaimergp) [inline recipe/meta.yaml]:
```suggestion
  url: https://github.com/conda/conda-libmamba-solver/archive/{{ version }}.tar.gz
```

I don't know what was temporary but the current URL doesn't look like it :D

**Review by jaimergp — CHANGES_REQUESTED** [review]
Leaving this here because we won't ship til Monday to have a calm weekend 🏖️ 🙏

**Review by jaimergp — APPROVED** [review]

**Comment by jaimergp** [comment]
Let's do this.

**Pull request merged**